modules/appeals_api/app/models/concerns/appeals_api/add_error.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module AddError
+    extend ActiveSupport::Concern
+
+    included do
+      def add_error(message, missing_fields: [], title: 'Missing fields', pointer: nil, code: nil)
+        error_options = { source: { pointer: pointer },
+                          detail: message,
+                          title: title,
+                          code: code,
+                          meta: { missing_fields: missing_fields } }
+        errors.add(:base, message, error_options)
```

Comment:
Feel free to extract some of the repeated validations into this concern as well! Some `add_future_date_error(pointer, meta)` or somesuch might be nice. Not a blocker, though, just pointing it out :)